### PR TITLE
Application load balancer quoin

### PR DESCRIPTION
**Pull request by bhague1281**
Similar to the ELB, with the following additions:
* Create an access log S3 bucket
* Option to enable deletion protection

### CODE REVIEW

File: alb-external/main.tf [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+resource "aws_lb" "external" {
+  name            = "${var.name}"
+  internal        = false
+  subnets         = ["${split(",", var.subnet_ids)}"]
+  security_groups = ["${aws_security_group.external_alb.id}"]
+
+  idle_timeout    = 60
+
+  enable_deletion_protection = "${var.enable_deletion_protection}"
+
```
Comment (weitenghuang) [inline alb-external/main.tf]: should we have [`enable_cross_zone_load_balancing`](enable_cross_zone_load_balancing)?
@dragan , I am not sure why TF docs says `This is a network load balancer feature` while it should be available for all types of balancers (application, network, or classic)

File: alb-external/main.tf [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port   = 443
```
Comment (weitenghuang) [inline alb-external/main.tf]: would this be the same value as `"${var.target_group_port}"`?

Comment (bhague1281) [reply alb-external/main.tf]: It's the same value as `var.listener_port`. Do you think it'd be better to have 1 ingress rule that's defined by `var.listener_port`?

Comment (weitenghuang) [reply alb-external/main.tf]: I think so. This security group is used by your alb which is listening at `var.listener_port`

File: alb-external/main.tf [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+* ------------------------------------------------------------------------------
+*/
+
+resource "aws_security_group" "external_alb" {
+  name        = "${format("%s-external-alb", var.name)}"
+  vpc_id      = "${var.vpc_id}"
+  description = "Allows external ALB traffic"
+
+  ingress {
+    from_port   = 80
```
Comment (weitenghuang) [inline alb-external/main.tf]: do you need port 80?

File: alb-external/main.tf (hunk 1 above)
Comment (weitenghuang) [reply alb-external/main.tf]: Got it. `With Application Load Balancers, cross-zone load balancing is always enabled.`
Ignore my previous comment.

**Comment by cfreiner** [comment]
Are we still waiting on any changes to this PR?

**Comment by nicklathe** [comment]
Please squash and sign the commit with GPG.

**Comment by nicklathe** [comment]
Also please use the signed-off-by feature. https://git-scm.com/docs/git-commit#git-commit---signoff

**Pull request merged**